packages/venia-ui/lib/components/CartPage/PriceAdjustments/GiftOptions/giftOptions.js

```diff
@@ -109,11 +109,8 @@ export const giftOptionsResolvers = {
                 id: cart_id
             });
 
-            const {
-                include_gift_receipt,
-                include_printed_card,
-                gift_message
-            } = cache.data.data[cartIdInCache];
+            const { include_gift_receipt, include_printed_card, gift_message } =
+                cache.data.data[cartIdInCache] || {};
```

Comment:
I wonder what GraphQL suggests for client side caching/resolvers for this situation. We will likely run into it elsewhere - I wonder if there is some sort of "reset callback" that can reset the store to an initial state. After all, we call `resetStore` on sign out which was effectively deleting the cache, causing this to throw. Maybe `resetStore` takes a callback?

This is fine, by the way - just postulating.

Reply:
Oh hey: https://www.apollographql.com/docs/react/caching/cache-interaction/#resetting-the-store

Reply:
Yes, I tried that a while ago when I was developing gift options. The problem is our gift options are name spaced to a certain cart id. Reset store, at that time, would not know which cart id to use to reset the store with empty gift options.

Reply:
Oh yeah @revanth0212 I forgot about that. So instead of doing it on store reset we would have to know to set that initial state whenever a cart is created. Hmm interesting! I don't think we can do that with the way that carts are created right now.